ec/src/msm/variable_base.rs

```diff
@@ -36,42 +38,36 @@ impl VariableBaseMSM {
                 let mut res = zero;
                 // We don't need the "zero" bucket, so we only have 2^c - 1 buckets
                 let mut buckets = vec![zero; (1 << c) - 1];
-                scalars
-                    .iter()
-                    .zip(bases)
-                    .filter(|(s, _)| !s.is_zero())
-                    .for_each(|(&scalar, base)| {
-                        if scalar == fr_one {
-                            // We only process unit scalars once in the first window.
-                            if w_start == 0 {
-                                res.add_assign_mixed(base);
-                            }
-                        } else {
-                            let mut scalar = scalar;
+                scalars_and_bases.clone().for_each(|(&scalar, base)| {
```

Comment:
Why is this cloning the vector? Seems like this vector isn't mutated

Reply:
It's not a vector, it's an iterator that contains only a pointer and an index, so it's cheap to clone.

Reply:
I see. Perhaps this should be renamed to `scalars_and_bases_iter`?